Model/Resolver/OrderTotal.php

```diff
@@ -0,0 +1,209 @@
+            'total_shipping' => ['value' => $order->getShippingAmount(), 'currency' => $currency],
+            'shipping_handling' => [
+                'amount_excluding_tax' => [
+                    'value' => $order->getShippingAmount(),
+                    'currency' => $order->getOrderCurrencyCode()
+                ],
+                'amount_including_tax' => [
+                    'value' => $order->getShippingInclTax(),
+                    'currency' => $currency
+                ],
+                'total_amount' => [
+                    'value' => $order->getShippingAmount(),
+                    'currency' => $currency
+                ],
+                'taxes' => $this->getAppliedShippingTaxesDetails($order),
+                'discounts' => $this->getShippingDiscountDetails($order),
+            ],
+            'model' => $order
+        ];
+    }
+
+    /**
+     * Retrieve applied taxes that apply to the order
+     *
+     * @param OrderInterface $order
+     * @return array
+     */
+    private function getAllAppliedTaxesOnOrders(OrderInterface $order): array
+    {
+        $extensionAttributes = $order->getExtensionAttributes();
+        $appliedTaxes = $extensionAttributes->getAppliedTaxes() ?? [];
+        $allAppliedTaxOnOrders = [];
+        foreach ($appliedTaxes as $taxIndex => $appliedTaxesData) {
+            $allAppliedTaxOnOrders[$taxIndex] = [
+                'title' => $appliedTaxesData->getDataByKey('title'),
+                'percent' => $appliedTaxesData->getDataByKey('percent'),
+                'amount' => $appliedTaxesData->getDataByKey('amount'),
+            ];
+        }
+        return $allAppliedTaxOnOrders;
+    }
+
+    /**
+     * Return taxes applied to the current order
+     *
+     * @param OrderInterface $order
+     * @return array
+     */
+    private function getAppliedTaxesDetails(OrderInterface $order): array
+    {
+        $allAppliedTaxOnOrders = $this->getAllAppliedTaxesOnOrders($order);
+        $taxes = [];
+        foreach ($allAppliedTaxOnOrders as $appliedTaxes) {
+            $appliedTaxesArray = [
+                'rate' => $appliedTaxes['percent'] ?? 0,
+                'title' => $appliedTaxes['title'] ?? null,
+                'amount' => [
+                    'value' => $appliedTaxes['amount'] ?? 0,
+                    'currency' => $order->getOrderCurrencyCode()
+                ]
+            ];
+            $taxes[] = $appliedTaxesArray;
+        }
+        return $taxes;
+    }
+
+    /**
+     * Return information about an applied discount
+     *
+     * @param OrderInterface $order
+     * @return array
+     */
+    private function getDiscountDetails(OrderInterface $order): array
+    {
+        $orderDiscounts = [];
+        if (!($order->getDiscountDescription() === null && $order->getDiscountAmount() == 0)) {
+            $orderDiscounts[] = [
+                'label' => $order->getDiscountDescription(),
```

Comment:
as I understand, this is the main part that got changed? If so, please add a note to the method docs